pyccel/codegen/printing/ccode.py

```diff
@@ -1045,6 +1045,36 @@ def _print_LiteralString(self, expr):
                                .replace("'", "\\'")
         return '"{}"'.format(format_str)
 
+    def _undo_print_LiteralString(self, expr):
+        """
+        Undo the changes done by _print_LiteralString() function on the given string.
+
+        This is required so that escape characters are displayed correctly in the string.
```

Comment:
Why do these changes need undoing? Do you have an example where the result was strange?

Is it possible that `_print_LiteralString` is being called on the print format string leading to the escape sequences being applied twice to literal arguments?